File: bm_db_android_java/app/src/main/java/steffen/de/bm_db_android_java/Controller/MainActivityController.java
package steffen.de.bm_db_android_java.Controller;

import android.content.Context;
import android.os.AsyncTask;

import java.util.ArrayList;
import java.util.List;

import steffen.de.bm_db_android_java.Database.DataDatabase;
import steffen.de.bm_db_android_java.Model.Data;

public class MainActivityController {

    private List<Data> dataList;
    private DataDatabase database;

    public MainActivityController(Context context){
        database = DataDatabase.getInstance(context);
        dataList = new ArrayList<Data>();

        addDataToList();
    }

    public void loadData(){
        long startTime = System.nanoTime();

        dataList = database.dataDao().getAll();

        long endTime = System.nanoTime();
        System.out.println("Time: Load -> " + ((endTime - startTime) / 1000000));
        System.out.println("Time: Load -> " + dataList.size());
    }

    public void resetList(){
        dataList.clear();
    }

    public void insertData(){
        long startTime = System.nanoTime();

        database.dataDao().insert(dataList);

        long endTime = System.nanoTime();
        System.out.println("Time: Insert -> " + ((endTime - startTime) / 1000000));
        System.out.println("Time: Insert -> " + database.dataDao().getAll().size());
    }

    public void deleteData(){
        long startTime = System.nanoTime();

        for (Data d : database.dataDao().getAll()) {
            database.dataDao().delete(d);
        }

        long endTime = System.nanoTime();
        System.out.println("Time: Delete -> " + ((endTime - startTime) / 1000000));
        System.out.println("Time: Delete -> " + database.dataDao().getAll().size());

        dataList.clear();
        addDataToList();
    }

    public void addDataToList()
    {
        for(int i = 0; i < 1000; i++){
            dataList.add(new Data());
        }
    }
}

File: bm_db_android_kotlin/app/src/main/java/steffen/de/bm_db_android_kotlin/Controller/MainActivityController.kt
package steffen.de.bm_db_android_kotlin.Controller

import android.app.Application
import kotlinx.coroutines.experimental.launch
import kotlinx.coroutines.experimental.runBlocking
import steffen.de.bm_db_android_kotlin.Model.Data
import steffen.de.bm_db_android_kotlin.Database.DataDatabase
import kotlin.system.measureTimeMillis

class MainActivityController(application: Application, private var database: DataDatabase = DataDatabase.getDatabase(application)){

    private var dataList: MutableList<Data>? = mutableListOf<Data>()

    init {
        for(i in 1..1000){
            dataList!!.add(Data())
        }
    }

    fun loadData() = runBlocking{

        val timer = measureTimeMillis {
            val job = launch{
                dataList = database.dataDao().all
            }
            job.join()
        }

        println("LoadTime:" + timer)
        println("Datansatz:" + dataList!!.count())
    }

    fun resetList(){
        dataList = null
    }

    fun insertData() = runBlocking{
        val timer = measureTimeMillis {
            val job = launch {
                for (d in dataList!!) {
                    database.dataDao().insert(d)
                }
            }
            job.join()
        }

        println("InsertTime: " + timer)
    }

    fun deleteData() = runBlocking{
        val timer = measureTimeMillis {
            val job = launch {
                for (d in database.dataDao().all) {
                    database.dataDao().delete(d)
                }
            }
            job.join()
        }

        println("DeleteTime: ")

        dataList!!.clear()
        for(i in 1..1000){
            dataList!!.add(Data())
        }


        println("Delete erfolgreich.")
    }

}